### PR TITLE
#3391 multiViewState for dataList

### PR DESCRIPTION
**Pull request by christophs78**
The first time i try to contribute code to an open source - project....

I looked into the implemention of MultiViewState in DataTable und tried to replicate the necessary part´s for DataList.
Like to hear your feedback and try to improove if necessary. :-)

### CODE REVIEW

File: src/main/java/org/primefaces/util/Constants.java [hunk 1]
```diff
@@ -88,6 +88,8 @@
 
     public static final String TABLE_STATE = "primefaces.TABLE_STATE";
 
+    public static final String LIST_STATE = "primefaces.LIST_STATE";
+
```
Comment (melloware) [inline src/main/java/org/primefaces/util/Constants.java:92]: Maybe DATALIST_STATE instead as LIST_STATE might be a little too ambiguous?

Comment (christophs78) [reply src/main/java/org/primefaces/util/Constants.java:92]: Good point. Renamed the constant.

**Comment by Rapster** [comment]
By the way, thank you for contributing

**Comment by melloware** [comment]
Yes thanks for contributing.  Would like @mertsincan to review this as well.

**Comment by christophs78** [comment]
org.primefaces.PrimeFaces has two methods concerning TableState
 
```
   public void clearTableStates() {
        getFacesContext().getExternalContext().getSessionMap().remove(Constants.TABLE_STATE);
    }

    public void clearTableState(String key) {
        Map<String, Object> sessionMap = getFacesContext().getExternalContext().getSessionMap();
        Map<String, TableState> dtState = (Map) sessionMap.get(Constants.TABLE_STATE);
        if (dtState != null) {
            dtState.remove(key);
        }
    }

```
I don´t find references to this methods. Should we offer the same for DATALIST_STATE?

**Comment by Rapster** [comment]
Please consider few comments I wrote (at least 2)

**Comment by christophs78** [comment]
> Please consider few comments I wrote (at least 2)

@Rapster - Perhaps dummy-question: Where do i see your comments? They do not seem to show up in the conversation of this pull-request.

**Comment by cnsgithub** [comment]
Perhaps something get lost yesterday during GitHub's maintenance work?

**Comment by tandraschko** [comment]
Your PR contains now 44 file changes, thats not cool :D

**Comment by christophs78** [comment]
> 
> 
> Your PR contains now 44 file changes, thats not cool :D

Tried to rebase on current master... Something did not work the way i expected. :-(

**Comment by tandraschko** [comment]
maybe someone like @melloware can assist?
i'm also a PR noob :D

**Comment by christophs78** [comment]
If necessary i may create a new "clean" PR.
Hoped to make life easier for you by rebasing but did not worked out.

**Comment by Rapster** [comment]
Reset your branch to head, and then do the rebase again (not a pull). I feel something went wrong during the process, conflicts?

**Comment by tandraschko** [comment]
second PR arrived, lets close this one

**Pull request closed without merging**